app/models/import.rb

```diff
@@ -1,14 +1,20 @@
 class Import < ApplicationRecord
+  FIELDS = [ :date, :name, :category, :tags, :amount ].freeze
+  REQUIRED_FIELDS =  [ :date, :amount ].freeze
+
   belongs_to :account
+  has_many :rows, dependent: :destroy, class_name: "Import::Row"
+
+  accepts_nested_attributes_for :rows
 
   validate :raw_file_must_be_parsable
   validates :col_sep, inclusion: { in: Csv::COL_SEP_LIST }
 
-  before_save :initialize_csv, if: :should_initialize_csv?
+  before_save :recreate_rows, if: :should_recreate_rows?
 
   enum :status, { pending: "pending", complete: "complete", importing: "importing", failed: "failed" }, validate: true
 
-  store_accessor :column_mappings, :define_column_mapping_keys
+  store :column_mappings, accessors: FIELDS, suffix: :column_mapping
```

Comment:
I think we'll want to leverage `store_accessor` here since we're using Postgres.  This allows us to offload all the serialization/deserialization to the DB rather than doing it in Rails.  

From the docs:

> NOTE: If you are using structured database data types (e.g. PostgreSQL hstore/json, or MySQL 5.7+ json) there is no need for the serialization provided by [.store](https://api.rubyonrails.org/classes/ActiveRecord/Store/ClassMethods.html#method-i-store). Simply use [.store_accessor](https://api.rubyonrails.org/classes/ActiveRecord/Store/ClassMethods.html#method-i-store_accessor) instead to generate the accessor methods. Be aware that these columns use a string keyed hash and do not allow access using a symbol.

https://api.rubyonrails.org/classes/ActiveRecord/Store.html